src/components/Views/CameraView/CameraView.js

```diff
@@ -87,12 +109,18 @@ class CameraView extends React.Component<
   }
 
   camera: RNCamera;
+  rightDrawer: Drawer;
 
   takePicture = async () => {
     const {
-      updateObservation,
+      createObservation,
+      fromEditor,
```

Comment:
I think you can just check if a `selectedObservation` exists, if it does then create the observation, otherwise update the selectedObservation